pkg/controller/volume/expand/expand_controller_test.go

```diff
@@ -0,0 +1,182 @@
+			pvc:             getFakePersistentVolumeClaim("good-pvc", "vol-3", "resizable2", "good-pvc-vol-3"),
+			storageClass:    getFakeStorageClass("resizable2", csitranslationplugins.AWSEBSInTreePluginName),
+			pvcKey:          "default/good-pvc",
+			expansionCalled: true,
+		},
+		{
+			name:                "when csi migration is enabled for a in-tree plugin",
+			csiMigrationEnabled: true,
+			pv:                  getFakePersistentVolume("vol-4", csitranslationplugins.AWSEBSInTreePluginName, "csi-pvc-vol-4"),
+			pvc:                 getFakePersistentVolumeClaim("csi-pvc", "vol-4", "resizable3", "csi-pvc-vol-4"),
+			storageClass:        getFakeStorageClass("resizable3", csitranslationplugins.AWSEBSInTreePluginName),
+			pvcKey:              "default/csi-pvc",
+		},
+	}
+
+	for _, tc := range tests {
+		test := tc
+		if tc.pv != nil {
+			informerFactory.Core().V1().PersistentVolumes().Informer().GetIndexer().Add(tc.pv)
+		}
+
+		if tc.pvc != nil {
+			informerFactory.Core().V1().PersistentVolumeClaims().Informer().GetIndexer().Add(tc.pvc)
+		}
+		allPlugins := []volume.VolumePlugin{}
+		allPlugins = append(allPlugins, awsebs.ProbeVolumePlugins()...)
+		if tc.storageClass != nil {
+			informerFactory.Storage().V1().StorageClasses().Informer().GetIndexer().Add(tc.storageClass)
+		}
+		expc, err := NewExpandController(fakeKubeClient, pvcInformer, pvInformer, storageClassInformer, nil, allPlugins)
+		if err != nil {
+			t.Fatalf("error creating expand controller : %v", err)
+		}
+
+		if test.csiMigrationEnabled {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, true)()
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationAWS, true)()
+		}
+
+		var expController *expandController
+		expController, _ = expc.(*expandController)
+		var expansionCalled bool
+		expController.operationGenerator = operationexecutor.NewFakeOgCounter(func() (error, error) {
+			expansionCalled = true
+			return nil, nil
+		})
+
+		err = expController.syncHandler(test.pvcKey)
+		if err != nil && !test.hasError {
+			t.Fatalf("for: %s; unexpected error while running handler : %v", test.name, err)
+		}
+		if expansionCalled != test.expansionCalled {
+			t.Fatalf("for: %s; expected expansionCalled to be %v but was %v", test.name, test.expansionCalled, expansionCalled)
+		}
+	}
+}
+
+func getFakePersistentVolume(volumeName, pluginName string, pvcUID types.UID) *v1.PersistentVolume {
+	pv := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{Name: volumeName},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeSource: v1.PersistentVolumeSource{},
+			ClaimRef: &v1.ObjectReference{
+				Namespace: "default",
+			},
+		},
+	}
+	if pvcUID != "" {
+		pv.Spec.ClaimRef.UID = pvcUID
+	}
+
+	if matched, _ := regexp.MatchString(`csi`, pluginName); matched {
+		pv.Spec.PersistentVolumeSource.CSI = &v1.CSIPersistentVolumeSource{
```

Comment:
This code path is IMO not executed. It would be nice to have test for a generic CSI (non-migrated) volume too.